src/GafferOSL/ShadingEngine.cpp

```diff
@@ -842,6 +842,22 @@ class ShadingResults
 
 };
 
+struct ThreadInfo
```

Comment:
It feels like this could also own a `ShadingContext *`, and make ShadingContextWrapper redundant. Does that make sense?